CONTRIBUTING.rst

```diff
@@ -36,6 +36,19 @@ Docker-based development
    ``Ctrl-D`` on OS X or Linux. Alternatively, you could also type the Python
    command ``exit()`` and press ``Enter``.
 
+   Set these environment variables::
+
+    SF_CALLBACK_URL='http://localhost:8080/accounts/salesforce-production/login/callback/'
+    SF_CLIENT_KEY='-----BEGIN RSA PRIVATE KEY-----[some key here]-----END RSA PRIVATE KEY-----'
+    SF_CLIENT_ID=...
+    SF_CLIENT_SECRET=...
+
+   The ``SF_CLIENT_ID`` and ``SF_CLIENT_SECRET`` values should be the
+   same as the values you will eventually put in the database when you
+   initialize the SocialApps. The value for ``SF_CLIENT_KEY`` can be
```

Comment:
Hmm, I don't see any instructions to run populate_social_apps in here?

For our other projects we have been intending to add a migration step that populates the social apps so that a separate command doesn't need to be run. Can you go ahead and do that for MetaShare?

Reply:
That's because the `populate_social_apps` instructions were already in the document. I've added to them.

I think that adding setup data to tables is a wrong thing to do via a Django migration; can you walk me through the rationale a bit? (In particular, as it's something that reasonably differs between deployments, including it in the migrations makes the set up brittle and over-specific, I think.)

Reply:
The rationale is that we always want them there, so it's frustrating to have an extra step which we sometimes forget to run.

The differences between deployments come down to the client id/secret which now default to the environment variables, right? Am I forgetting some scenario where we don't want them there?

Reply:
So you're saying that the migration should pull the values from the env? That leaves me feeling really uncomfortable, but I can't articulate why.

Reply:
Yeah. I mean really I'd like them to come from the env at runtime rather than migration time, but since that's not really how django-allauth works, this at least gets us a step closer. We still need to remember to update the values in the db if we change them in the environment, but this would at least remove one time that we have to do that.

Reply:
The GitHub values are not currently pulled from any env vars. I tend to agree with Kit on what seems cleaner here from a theoretical perspective, but I understand the practical convenience of adding a setup migration. It does seem like a separate PR for that change would be fine, and it doesn't need to hold up this merge.

Reply:
I think part of why I am uncomfortable with this is that it makes applying migrations non-repeatable; the resultant DB state will depend on a _very literal_ environmental factor. If you're OK with that, we can do it, but I think it's generally worth avoiding.

Reply:
What's so repeatable about migrations to begin with? Their behavior already generally depends on what data is in the db and changes over time as more migrations are added. So I'm not sure I understand what design goal you're aiming for.

Reply:
It's not important.